### ISSUE TITLE
bug reported by Zim

### ISSUE THREAD

**Issue by tuxedotje** [zim-desktop-wiki/zim-desktop-wiki#1212]
This is zim 0.73.0
Platform: posix
Locale: en_US UTF-8
FS encoding: utf-8
Python: (3, 9, 0, 'beta', 4)
PyGObject: (3, 36, 1)

======= Traceback =======
  File "/usr/lib/python3.9/site-packages/zim/gui/pageview.py", line 5622, in set_page
    tree = self.notebook.get_template(page)
  File "/usr/lib/python3.9/site-packages/zim/notebook/notebook.py", line 1123, in get_template
    return self.eval_new_page_template(path, template)
  File "/usr/lib/python3.9/site-packages/zim/notebook/notebook.py", line 1145, in eval_new_page_template
    return parser.parse(lines)
  File "/usr/lib/python3.9/site-packages/zim/formats/wiki.py", line 631, in parse
    mywikiparser(builder, input)
  File "/usr/lib/python3.9/site-packages/zim/formats/wiki.py", line 190, in __call__
    builder.start(FORMATTEDTEXT)
  File "/usr/lib/python3.9/site-packages/zim/formats/__init__.py", line 783, in start
    self._b.start(tag, attrib)
TypeError: start() argument 2 must be dict, not None

**Comment by kapsh** [comment]
Another example: 
```
This is zim 0.73.2
Platform: posix
Locale: en_GB UTF-8
FS encoding: utf-8
Python: (3, 9, 0, 'candidate', 1)
PyGObject: (3, 36, 0)

======= Traceback =======
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/gui/pageview.py", line 5617, in set_page
    tree = page.get_parsetree()
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/notebook/page.py", line 568, in get_parsetree
    self._parsetree = parser.parse(text, file_input=True)
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/formats/wiki.py", line 639, in parse
    mywikiparser(builder, input)
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/formats/wiki.py", line 191, in __call__
    self.block_parser(builder, text)
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/parser.py", line 437, in __call__
    self._raise_exception(error, text, mstart, mend, builder, self.rules[i])
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/parser.py", line 435, in __call__
    self.rules[i].process(builder, *groups)
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/formats/wiki.py", line 329, in parse_pre
    builder.append(VERBATIM_BLOCK, attrib, text)
  File "/usr/x86_64-pc-linux-gnu/lib/python3.9/site-packages/zim/formats/__init__.py", line 833, in append
    self._b.start(tag, attrib)
TypeError: start() argument 2 must be dict, not None
```

Happens only when running under python 3.9. And I think this was caused by notebook entry with preformatted text (code).

**Comment by kapsh** [comment]
@tuxedotje your issue should be fixed in 0.73.2 by commit feba6b2cf2b23c4c019a7bc770da44064cc6dd42

**Comment by introt** [comment]
Please open a new issue if the problem persists on the latest version. Thank you!
